mmpose/models/mesh_heads/discriminator.py

```diff
@@ -0,0 +1,289 @@
+        len_channels = len(channels)
+        for idx in range(len_channels - 2):
+            self.conv_blocks.add_module(
+                name='conv_{}'.format(idx),
+                module=nn.Conv2d(
+                    in_channels=channels[idx],
+                    out_channels=channels[idx + 1],
+                    kernel_size=1,
+                    stride=1))
+
+        self.fc_layer = nn.ModuleList()
+        for idx in range(joint_count):
+            self.fc_layer.append(
+                nn.Linear(
+                    in_features=channels[len_channels - 2], out_features=1))
+
+    def forward(self, inputs):
+        """Forward function.
+
+        The input is (batch size x joint_count x 9)
+        """
+        # batch_size = inputs.shape[0]
+        inputs = inputs.transpose(1, 2).\
+            unsqueeze(2).contiguous()  # to N x 9 x 1 x joint_count
+        internal_outputs = self.conv_blocks(
+            inputs)  # to N x c x 1 x joint_count
+        outputs = []
+        for idx in range(self.joint_count):
+            outputs.append(self.fc_layer[idx](internal_outputs[:, :, 0, idx]))
+
+        return torch.cat(outputs, 1), internal_outputs
+
+
+class FullPoseDiscriminator(LinearModel):
+    """Discriminator for SMPL pose parameters of all joints.
+
+    Args:
+        fc_layers (List): List of neuron count,
+         such as [736, 1024, 1024, 1]
+        use_dropout (List): List of bool define use dropout or not
+         for each layer, such as [True, True, False]
+        drop_prob (List): List of float defined the drop prob,
+         such as [0.5, 0.5, 0]
+        use_ac_func(List): List of bool define use active
+         function or not, such as [True, True, False]
+    """
+
+    def __init__(self, fc_layers, use_dropout, drop_prob, use_ac_func):
+        if fc_layers[-1] != 1:
+            msg = 'the neuron count of the last layer must be 1,' \
+                  ' but got {}'.format(fc_layers[-1])
+            raise ValueError(msg)
+
+        super().__init__(fc_layers, use_dropout, drop_prob, use_ac_func)
+
+    def forward(self, inputs):
+        """Forward function."""
+        return self.fc_blocks(inputs)
+
+
+class SMPLDiscriminator(nn.Module):
+    """Discriminator for SMPL pose and shape parameters. It is composed of a
+    discriminator for SMPL shape parameters, a discriminator for SMPL pose
+    parameters of all joints  and a discriminator for SMPL pose parameters of
+    each joint.
+
+    Args:
+        beta_channel (list of int): List of neuron count of the
+            discriminator of shape parameters. Defaults to [10, 5, 1]
+        per_joint_channel (list of int): List of neuron count of the
+            discriminator of each joint. Defaults to [9, 32, 32, 1]
+        full_pose_channel (list of int): List of neuron count of the
+            discriminator of full pose. Defaults to [23*32, 1024, 1024, 1]
+    """
+
+    def __init__(self,
+                 beta_channel=[10, 5, 1],
```

Comment:
do not use list as default args.

tuple is acceptable

Reply:
OK, I will modify the code.